x-pack/test/observability_ai_assistant_functional/tests/feature_controls/settings_security.spec.ts

```diff
@@ -60,22 +61,68 @@ export default function ({ getPageObjects, getService }: FtrProviderContext) {
+          ensureCurrentUrl: false,
+          shouldLoginIfPrompted: false,
+          shouldUseHashForSubUrl: false,
+        });
+        const testSearchConnectorIndexPattern = 'my-search-index-pattern';
         const searchConnectorIndexPatternInput = await testSubjects.find(
           ui.pages.settings.searchConnectorIndexPatternInput
         );
+        // make sure the input is empty (default value)
         await searchConnectorIndexPatternInput.clearValue();
         await searchConnectorIndexPatternInput.type(testSearchConnectorIndexPattern);
         const saveButton = await testSubjects.find(ui.pages.settings.saveButton);
         await saveButton.click();
-        await browser.refresh();
-        const searchConnectorIndexPatternInputValue =
-          await searchConnectorIndexPatternInput.getAttribute('value');
-        expect(searchConnectorIndexPatternInputValue).to.be(testSearchConnectorIndexPattern);
-        // reset the value
-        await searchConnectorIndexPatternInput.clearValue();
-        await searchConnectorIndexPatternInput.type('logs-*');
+        // wait for page to refrsh
+        await testSubjects.missingOrFail(ui.pages.settings.searchConnectorIndexPatternInput, {
+          timeout: 2000,
+        });
+        // wait for the new page to fully load
+        await testSubjects.existOrFail(ui.pages.settings.searchConnectorIndexPatternInput, {
+          timeout: 2000,
+        });
+        expect(await searchConnectorIndexPatternInput.getAttribute('value')).to.be(
+          testSearchConnectorIndexPattern
+        );
+        // reset the value back to default
+        const resetToDefaultLink = await testSubjects.find(ui.pages.settings.resetToDefaultLink);
+        await resetToDefaultLink.click();
+
+        expect(await searchConnectorIndexPatternInput.getAttribute('value')).to.be('');
         await saveButton.click();
+        await testSubjects.missingOrFail(ui.pages.settings.searchConnectorIndexPatternInput, {
+          timeout: 2000,
+        });
+        // wait for the new page to fully load
+        await testSubjects.existOrFail(ui.pages.settings.searchConnectorIndexPatternInput, {
+          timeout: 2000,
+        });
+        expect(await searchConnectorIndexPatternInput.getAttribute('value')).to.be('');
+      });
+      it('displays failure toast on failed request', async () => {
+        const searchConnectorIndexPatternInput = await testSubjects.find(
+          ui.pages.settings.searchConnectorIndexPatternInput
+        );
+        await searchConnectorIndexPatternInput.clearValue();
+        await searchConnectorIndexPatternInput.type('test');
+
+        await interceptRequest(
+          driver.driver,
+          '*kibana\\/settings*',
+          (responseFactory) => {
+            return responseFactory.fail();
+          },
+          async () => {
+            await testSubjects.click(ui.pages.settings.saveButton);
+          }
```

Comment:
Instead of having a callback, can we have `interceptRequest` return the value? Also, using destructured args will make it easier to read.

```suggestion
        await interceptRequest({ driver: driver.driver, pattern: '*kibana\\/settings*' }, (responseFactory) => {
          return responseFactory.fail();
        }
        await testSubjects.click(ui.pages.settings.saveButton);
```